src/picosvg/svg.py

```diff
@@ -381,111 +479,154 @@ def _combine_clip_paths(self, clip_paths: Sequence[SVGPath]) -> SVGPath:
             return clip_paths[0]
         return SVGPath.from_commands(intersection(clip_paths))
 
-    def _new_id(self, tag, template):
-        for i in range(100):
+    def _new_id(self, template):
+        for i in range(1 << 16):
             potential_id = template % i
-            existing = self.xpath(f'//svg:{tag}[@id="{potential_id}"]')
+            existing = self.xpath(f'//svg:*[@id="{potential_id}"]')
             if not existing:
                 return potential_id
         raise ValueError(f"No free id for {template}")
 
-    def _ungroup(self, scope_el):
-        """Push inherited attributes from group down, then remove the group.
+    def breadth_first(self):
+        frontier = [
+            SVGTraverseContext(
+                0,
+                self.svg_root,
+                "/svg[0]",
+                Affine2D.identity(),
```

Comment:
technically the root `<svg>` element itself could also have a `transform` attribute, which would mean that the current transform is something else than identity?

Reply:
Having read that I'm wondering if we shouldn't file a bug specifically about that. However, I don't think we want to try to resolve that here.

Reply:
#228